src/Adapter/ParameterContainer.php

```diff
@@ -79,7 +84,15 @@ public function offsetExists($name)
      */
     public function offsetGet($name)
     {
-        return (isset($this->data[$name])) ? $this->data[$name] : null;
+        if (isset($this->data[$name])) {
+            return $this->data[$name];
+        }
+
+        if (isset($this->nameMapping[$name])) {
```

Comment:
I think it require a double check against `$this->data` to ensure no edge case : 

```suggestion
        if (isset($this->nameMapping[$name]) && isset($this->data[$this->nameMapping[$name]])) {
```